collect_app/src/main/java/org/odk/collect/android/widgets/items/SelectOneFromMapDialogFragment.kt

```diff
@@ -0,0 +1,137 @@
+import org.odk.collect.androidshared.livedata.MutableNonNullLiveData
+import org.odk.collect.androidshared.livedata.NonNullLiveData
+import org.odk.collect.androidshared.ui.FragmentFactoryBuilder
+import org.odk.collect.geo.MappableSelectItem
+import org.odk.collect.geo.SelectionMapData
+import org.odk.collect.geo.SelectionMapFragment
+import org.odk.collect.geo.SelectionMapFragment.Companion.REQUEST_SELECT_ITEM
+import org.odk.collect.material.MaterialFullScreenDialogFragment
+import javax.inject.Inject
+
+class SelectOneFromMapDialogFragment : MaterialFullScreenDialogFragment(), FragmentResultListener {
+
+    @Inject
+    lateinit var formEntryViewModelFactory: FormEntryViewModel.Factory
+    private val formEntryViewModel: FormEntryViewModel by activityViewModels { formEntryViewModelFactory }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        DaggerUtils.getComponent(context).inject(this)
+
+        childFragmentManager.fragmentFactory = FragmentFactoryBuilder()
+            .forClass(SelectionMapFragment::class.java) {
+                val formIndex = requireArguments().getSerializable(ARG_FORM_INDEX) as FormIndex
+                val prompt = formEntryViewModel.getQuestionPrompt(formIndex)
+                SelectionMapFragment(
+                    SelectChoicesMapData(prompt),
+                    skipSummary = true,
+                    showNewItemButton = false
+                )
+            }
+            .build()
+
+        childFragmentManager.setFragmentResultListener(REQUEST_SELECT_ITEM, this, this)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        val binding = SelectOneFromMapDialogLayoutBinding.inflate(inflater)
+        return binding.root
+    }
+
+    override fun getToolbar(): Toolbar? {
+        return null
+    }
+
+    override fun onBackPressed() {
+        dismiss()
+    }
+
+    override fun onCloseClicked() {
+        // No toolbar so not relevant
+    }
+
+    override fun onFragmentResult(requestKey: String, result: Bundle) {
+        val selectedIndex = result.getLong(SelectionMapFragment.RESULT_SELECTED_ITEM).toInt()
+        val formIndex = requireArguments().getSerializable(ARG_FORM_INDEX) as FormIndex
+        val prompt = formEntryViewModel.getQuestionPrompt(formIndex)
+        val selectedChoice = prompt.selectChoices[selectedIndex]
+        formEntryViewModel.answerQuestion(formIndex, SelectOneData(selectedChoice.selection()))
+        dismiss()
+    }
+
+    companion object {
+        const val ARG_FORM_INDEX = "form_index"
+    }
+}
+
+internal class SelectChoicesMapData(prompt: FormEntryPrompt) : SelectionMapData {
+
+    private val mapTitle = MutableLiveData(prompt.longText)
+    private val itemCount = MutableLiveData<Int>()
+    private val items = MutableNonNullLiveData(emptyList<MappableSelectItem>())
```

Comment:
Yes because they need to be returned as `LiveData` in the overriden getters. You'll see the form map case (`FormMapViewModels`) makes more use of the reactive values, and we'll probably want to expand that by adding progress dialogs/loading states to these views in the future.